test/collections.js

```diff
@@ -123,6 +123,11 @@ describe('Collections', function () {
       expect(map2).to.have.entries([['a', 'b'], ['c', 'd']]);
     });
 
+    it('non-entry objects should throw', function () {
+      expect(new Map([1, 2, 3])).to['throw'](TypeError);
```

Comment:
According to the [spec](http://tc39.github.io/ecma262/#sec-map-iterable), step 8, it simply calls `[0]` and `[1]` on the thing the iterable returns. In other words, this should be an iterable of three `undefined → undefined` entry pairs.